src/main/java/org/broadinstitute/hellbender/engine/spark/SAMRecordToGATKReadAdapterSerializer.java

```diff
@@ -86,12 +82,12 @@ public LazyBAMRecord(
 
         @Override
         public void setReferenceIndex(final int referenceIndex) {
-            mReferenceIndex = referenceIndex;
+            mReferenceIndex = null; // null out
```

Comment:
I don't think a warning would be helpful. The point is that the reference indices are being nulled out on the write side anyway (since the reads are headerless), so we know they have to be null on the read side.